shared/Docs/Navigation.tsx

```diff
@@ -175,43 +171,72 @@ function ActivePageMarker({ group, pathname }) {
 
 export function PageSidebar() {
   let isInsideMobileNavigation = useIsInsideMobileNavigation();
-  let [router, sections] = useInitialValue(
-    [useRouter(), useSectionStore((s) => s.sections)],
-    isInsideMobileNavigation
-  );
+  let router = useRouter();
+  let sections = useSectionStore((s) => s.sections);
+
+  let [pageSectionsEl, setPageSectionsEl] = useState(null);
+  let [pageSectionListItems, setPageSectionListItems] = useState(null);
+  let [windowWidth, setWindowWidth] = useState(null);
+
+  useEffect(() => {
+    const updateWindowWidth = () => setWindowWidth(window.innerWidth);
+    window.addEventListener("resize", updateWindowWidth);
+    return () => {
+      window.removeEventListener("resize", updateWindowWidth);
+    };
+  }, []);
+
+  useEffect(() => {
+    if (pageSectionsEl) {
+      setPageSectionListItems([
+        ...(pageSectionsEl?.querySelectorAll("li") ?? []),
+      ]);
+    }
+  }, [router.pathname, windowWidth, pageSectionsEl]);
```

Comment:
this is the "weird bit" -- see the "Code Quality" section in the [PR description](https://github.com/inngest/website/pull/730)